tutoraspects/templates/aspects/apps/aspects/migrations/alembic/versions/0021_completion_events.py

```diff
@@ -0,0 +1,65 @@
+"""
+create a top-level materialized view for completion events
+"""
+from alembic import op
+
+
+revision = "0021"
+down_revision = "0020"
+branch_labels = None
+depends_on = None
+on_cluster = " ON CLUSTER '{{CLICKHOUSE_CLUSTER_NAME}}' " if "{{CLICKHOUSE_CLUSTER_NAME}}" else ""
+engine = "ReplicatedReplacingMergeTree" if "{{CLICKHOUSE_CLUSTER_NAME}}" else "ReplacingMergeTree"
+
+
+def upgrade():
+    op.execute(
+        f"""
+        CREATE TABLE IF NOT EXISTS {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_COMPLETION_EVENTS_TABLE }} 
+        {on_cluster}
+        (
+            `event_id` UUID NOT NULL,
+            `emission_time` DateTime64 NOT NULL,
+            `actor_id` String NOT NULL,
+            `object_id` String NOT NULL,
+            `course_key` String NOT NULL,
+            `org` String NOT NULL,
+            `verb_id` LowCardinality(String) NOT NULL,
+            `progress_percent` Float32,
```

Comment:
This can be stored as an `Int32` as its value is between `1-100`

Reply:
Should probably make that change, then. Otherwise 👍

Reply:
Updated